atlp-rwanda/hackers-ec-be
# #187419059  implement two factor authentication

#### What does this PR do?
This PR implement the 2 factor authentication while use try to log in

#### Description of Task to be completed?

- To sent the email to the user to user who log in have otp and link containing the token direct him/her to the authentication where required to fill the otp received to the email
- authenticate the user based on otp

#### How should this be manually tested?

-clone the project
- run npm run install
- run npm run dev
- try login using this endpoint if you are locally and already sign up /api/v1/users/login
- check your email address you found the an email inbox have link na otp
- try to send the otp on our swagger doc we have the end point of 2FA api/v1/users/2fa/:token try to use link received on email and add otp

#### What are the relevant pivotal tracker/Trello stories?

#187419059
